docs/design_notes/module_lifecycle.md

```diff
@@ -0,0 +1,289 @@
+# The Module Lifecycle
```

Comment:
Suggestion: Use sentence case

```suggestion
# The Module lifecycle
```

Reply:
The only reason IMO we should use sentence case, is because when you have Flax names that use common English names—such as `Module`—titles may not make it obvious to users.

So, here:

```suggestion
# The Module lifecycle
```

it's more apparent that `Module` is a name, and not just a word, such as "lifecycle". If @jheek introduces a class called `Lifecycle`, this title may get confusing (e.g. when users start searching for `Lifecycle` on Flax docs/search engines)

Reference: https://developers.google.com/style/capitalization?hl=en#capitalization-in-titles-and-headings

Reply:
I agree it creates some ambiguity between class names and normal "words"